src/main/java/com/google/sps/servlets/WaterPollutionServlet.java

```diff
@@ -0,0 +1,121 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Servlet that returns a list of all Superfund Sites */
+@WebServlet("/water")
+public class WaterPollutionServlet extends HttpServlet {
+
+    private static final Logger logger = LoggerFactory.getLogger(WaterPollutionServlet.class);
+    
+    public static final String EPA_WATERSYSTEM_LINK = "https://enviro.epa.gov/enviro/efservice/WATER_SYSTEM/";
+    public static final String EPA_CITY_PARAMETER = "CITIES_SERVED/CONTAINS/";
+    public static final String EPA_STATE_PARAMETER = "/PRIMACY_AGENCY_CODE/";
+    public static final String MIN_DATE = "/ENFDATE/>/01-JAN-14";
+    public static final String CSV_FORMAT = "/Excel/";
+    public static final String SPLITERATOR = "\",\"";
+    public static final int TOTAL_CELL_COUNT = 47;
+
+    private static final String AREA_PARAMETER = "area";
+    private static final String ZIP = "zip";
+    private static final String TOWN_PARAMATER = "town";
+    private static final String STATE_PARAMATER = "state";
+
+    private static final String ZIP_PARAMETER = "zip_code";
+
+    public static final int DEFAULT_SCORE = 100;
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String area = request.getParameter(AREA_PARAMETER);
+        response.setContentType("application/json");
+        // if(area.equals(ZIP)){
+        //     String json = new Gson().toJson(
+        //         retrieveSDWViolations(request.getParameter(ZIP_PARAMETER), EPA_ZIP_FORMAT));
+        //     response.getWriter().write(json);
+        // } else if(area.equals(STATE)) {
+        //     String json = new Gson().toJson(
+        //         retrieveSDWViolations(request.getParameter(ZIP_PARAMETER), EPA_STATE_FORMAT));
+        //     response.getWriter().write(json);
+        // }
+        String json = new Gson().toJson(
+            retrieveSDWViolations(request.getParameter(TOWN_PARAMATER), request.getParameter(STATE_PARAMATER)));
+        response.getWriter().write(json);
+    }
+
+    /**
+    * A function that returns an arraylist of all the superfund sites in 
+    * a given zip code
+    * @param areaCode the zip code or state to pull data for
+    * @param areaType the Zip or State format for the EPA URL
+    * @return the list of superfund sites pulled from the EPA API
+    */
+    public ArrayList<WaterSystem> retrieveSDWViolations(String town, String state){
+        ArrayList<WaterSystem> waterSystems;
+        try {
+            URL url = new URL(EPA_WATERSYSTEM_LINK+EPA_CITY_PARAMETER+town.toUpperCase()+EPA_STATE_PARAMETER+state + CSV_FORMAT);
+            // URL url = new URL(EPA_API_LINK + areaType + areaCode + MIN_DATE + CSV_FORMAT);
+            waterSystems = parseViolationsFromURL(url);
+        } catch (IOException e){
+            logger.error(e.getMessage());
+            waterSystems = new ArrayList<>();
```

Comment:
A better way to handle this error would be to trickle it back up to the caller (WaterPollutionServlet.java line 75), catch the error there and return a 500 (Internal Server Error) response back to the caller. You can move the error logging mechanism up there too.

Returning a 500 error makes it easier to see at the browser why certain pieces of data are not loading correctly, e.g. through Chrome's dev tools. As implemented, an error here will manifest as an empty response to the original request - not so helpful for debugging.